client/src/components/EditProfileDialog.js

```diff
@@ -0,0 +1,164 @@
+import React, { useState, useCallback } from "react";
+import axios from "axios";
+import {
+  Button,
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  DialogActions,
+  TextField,
+  Grid,
+  Typography,
+} from "@material-ui/core";
+
+import DropZoneUpload from "../components/DropZoneUpload";
+import ExpertiseChips from "../components/ExpertiseChips";
+
+function EditProfileDialog(props) {
+  // Used for initializing values
+  const user = props.user;
+
+  // State variables
+  const [dialogOpen, setDialogOpen] = useState(false);
+
+  const [location, setLocation] = useState(user.location);
+  const [description, setDescription] = useState(user.description);
+  const [expertise, setExpertise] = useState(user.expertise);
+  const [linkedin, setLinkedin] = useState(user.linkedin);
+  const [angelco, setAngelco] = useState(user.angelco);
+
+  const [upload, setUpload] = useState(false);
+
+  const handleOpenDialog = (event) => {
+    setDialogOpen(true);
+  };
+
+  const handleCloseDialog = (event) => {
+    setDialogOpen(false);
+  };
+
+  const handleUpdateExpertise = (expertise) => {
+    setExpertise(expertise);
+  };
+
+  const handleUpdateDescription = (event) => {
+    setDescription(event.target.value);
+  };
+
+  const handleUpdateLocation = (event) => {
+    setLocation(event.target.value);
+  };
+
+  const handleUpdateLinkedin = (event) => {
+    setLinkedin(event.target.value);
+  };
+
+  const handleUpdateAngelco = (event) => {
+    setAngelco(event.target.value);
+  };
+
+  const handleTriggerFileUpload = (event) => {
+    // Trigger photo upload to s3
+    setUpload(true);
+
+    setDialogOpen(false);
+  };
+
+  const handleSave = useCallback((profilePics) => {
```

Comment:
A small question, is "useCallback" necessary here? There's no dependencies, so it acts just like a function.

Reply:
The function is actually dependent on the state of the EditProfileDialog component, I just forgot to pass them in as dependencies. I will add those in.